### ISSUE TITLE
Handle duplicate device names/request device name at onboarding

### ISSUE THREAD

**Issue by TomBrien** [home-assistant/iOS#1555]
**Is your feature request related to a problem? Please describe.**

Putting in as a FR to suggest/discuss a solution for the long-standing issue #415.

Depending on how an iPhone etc. is set up it is raltively easy to just end up with the default "iPhone" device name. In households/instances with more than one device set up like this the we end up with `_2` sensors which isn't the worst but also only one device will receive `notify.mobile_app_iphone` notifications, presumably determined by a race condition on the token updating. 

**Describe the solution you'd like**
Ideally, during onboarding we should check the list from:
```python
names = [
    hass.data["mobile_app"][DATA_CONFIG_ENTRIES][id]["name"] for id in hass.data["mobile_app"][DATA_CONFIG_ENTRIES].keys()
]
```

and either register the new device as `_2` rather than duplicate name or better yet prompt for a more useful name. 

Perhaps it's better to just always prompt for a device name during onboarding? I believe this is the Android behaviour? (@dshokouhi is that right?)

**Additional context**

**Comment by dshokouhi** [comment]
> Perhaps it's better to just always prompt for a device name during onboarding? I believe this is the Android behaviour? (@dshokouhi is that right?)

Yup we show the device name before finishing onboarding so the user can change it if they want to.
